### DISCUSSION TITLE
New Demo: Dynamic Tooltip Content

### DISCUSSION THREAD

**Discussion by kumilingus** [clientIO/joint#2102] [News]
Today's demo shows how to **display the tooltip** when you **hover over a port**. The tooltip shows various info about the port connection (what it is connected to and how) and the **information is gathered on demand** from the graph just before the tooltip is displayed. Check out this [CodePen](https://codepen.io/jointjs/full/jOvRPGZ).

<img width="1223" alt="image" src="https://user-images.githubusercontent.com/3967880/228558489-a98aadc5-c05c-4b3a-9110-e676c69f658f.png">

**Comment by samiulla1111** [comment]
Hi, can you please show a demo of a customized tooltip with react+typescript?

**Reply by kumilingus** [reply]
Could you please elaborate? Not sure what you mean by "customized tooltip with React + Typescript".

**Comment by samiulla1111** [comment]
I want to implement a customized tooltip with react js. So, I was passing my custom tooltip component in the new joint. ui.Tooltip({}) something like this.

new joint.ui.Tooltip({
    rootTarget: document.body,
    target: [data-tooltip],
    padding: 15
    content: <CustomTooltip/> //Note : I know content only accepts string | Element | jQuery | function(element)
});

please provide an example to create a custom tooltip with react js.

**Reply by kumilingus** [reply]
If you want the content of the tooltip to be a React component then:

1. Use a React Tooltip component instead (e.g. https://github.com/ReactTooltip/react-tooltip)
2. Perhaps try to use [React.portals](https://legacy.reactjs.org/docs/portals.html)

Otherwise, I suggest defining the content of the tooltip as an HTML string or an [HTMLElement](https://developer.mozilla.org/en-US/docs/Web/API/Document/createElement).

**Comment by samiulla1111** [comment]
Rephrasing the query. Does jointjs or js+ support integrating a react component within jointJS tool tip? 

P.S.: HTML Element or String is not sufficient for my use case. I have some custom react components which I am looking forward to integrate with JointJS tooltips.


new joint.ui.Tooltip({
rootTarget: document.body,
target: [data-tooltip],
padding: 15
content: // [Can I integrate some react component here OR through some other way?]
});

**Reply by kumilingus** [reply]
The answer is still the same.

> If you want the content of the tooltip to be a React component then:
> 
> 1. Use a React Tooltip component instead (e.g. https://github.com/ReactTooltip/react-tooltip)
> 2. Perhaps try to use [React.portals](https://legacy.reactjs.org/docs/portals.html)
> 
> Otherwise, I suggest defining the content of the tooltip as an HTML string or an [HTMLElement](https://developer.mozilla.org/en-US/docs/Web/API/Document/createElement).
